pycytominer/cyto_utils/collate.py

```diff
@@ -118,43 +119,29 @@ def collate(
             print(f"Ingesting {input_dir}")
         run_check_errors(ingest_cmd)
 
+        # Create a sqlite3 connection
+        connection = sqlite3.connect(cache_backend_file, isolation_level=None)
+        cursor = connection.cursor()
+
         if column:
             if print:
                 print(f"Adding a Metadata_Plate column based on column {column}")
-            alter_cmd = [
-                "sqlite3",
-                cache_backend_file,
-                "ALTER TABLE Image ADD COLUMN Metadata_Plate TEXT;",
-            ]
-            run_check_errors(alter_cmd)
-            update_cmd = [
-                "sqlite3",
-                cache_backend_file,
-                f"UPDATE image SET Metadata_Plate ={column};",
-            ]
-            run_check_errors(update_cmd)
+            cursor.execute("ALTER TABLE Image ADD COLUMN Metadata_Plate TEXT;")
+            cursor.execute(f"UPDATE image SET Metadata_Plate ={column};")
 
         if printtoscreen:
             print(f"Indexing database {cache_backend_file}")
-        index_cmd_img = [
-            "sqlite3",
-            cache_backend_file,
-            "CREATE INDEX IF NOT EXISTS table_image_idx ON Image(TableNumber, ImageNumber);",
-        ]
-        run_check_errors(index_cmd_img)
+        cursor.execute(
+            "CREATE INDEX IF NOT EXISTS table_image_idx ON Image(TableNumber, ImageNumber);"
+        )
         for eachcompartment in ["Cells", "Cytoplasm", "Nuclei"]:
-            index_cmd_compartment = [
-                "sqlite3",
-                cache_backend_file,
-                f"CREATE INDEX IF NOT EXISTS table_image_object_{eachcompartment.lower()}_idx ON {eachcompartment}(TableNumber, ImageNumber, ObjectNumber);",
-            ]
-            run_check_errors(index_cmd_compartment)
-        index_cmd_metadata = [
-            "sqlite3",
-            cache_backend_file,
-            "CREATE INDEX IF NOT EXISTS plate_well_image_idx ON Image(Metadata_Plate, Metadata_Well);",
-        ]
-        run_check_errors(index_cmd_metadata)
+            cursor.execute(
+                f"""CREATE INDEX IF NOT EXISTS table_image_object_{eachcompartment.lower()}_idx 
+                            ON {eachcompartment}(TableNumber, ImageNumber, ObjectNumber);"""
+            )
+        cursor.execute(
+            "CREATE INDEX IF NOT EXISTS plate_well_image_idx ON Image(Metadata_Plate, Metadata_Well);"
+        )
```

Comment:
The main purpose for the transition from `subprocess.run(['sqlite3', ...])` to the `cursor.execute()` approach is because the `sqlite3` cli command isn't installed for some reason when using the poetry dependency manager. Instead of troubleshooting that issue, I figured it would be better to adopt a more pythonic approach. 

Despite the change in structure, I deliberately limited the changes execute the exact same SQL commands. I would prefer to address performance/storage considerations in the context of a different PR.

Reply:
To clarify the problems with the `subprocess.run()` access to command line tools, it seems to be a corner case that involves the use of VSCode's pytest integration, poetry, and cli commands from extras packages. I confirmed that all the required tools are still properly installed in the actual environment created by poetry. 

I also changed the `cytominer-database` integration from the cli to import form for the same reason.

Reply:
> Given the changes to collate.py I recommend including those details upfront in the PR title or description to help observe and reason through what historically changed in this PR.

This is done.

Reply:
Cheers, thanks for the added effort with implementing and documenting this here. Seeing these changes made an impression on me how much database management is happening in this area. To this effect, creating indexes and defragmentation is sometimes an unannounced expectation of audiences receiving data (often this is very much warranted to provide them the best possible chance of utility, among other things). All this said, I'm good with leaving things as is and greatly appreciate the movement towards the import-based methods.